server/extension/extension.go

```diff
@@ -44,6 +44,17 @@ const (
 	// Example:
 	//     Argocd-Project-Name: "default"
 	HeaderArgoCDProjectName = "Argocd-Project-Name"
+
+	// HeaderArgoCDTargetCluster defines the target cluster URL
+	// that the application is associated with. This header will
+	// be populated by the extension proxy and passed to the
+	// configured backend service. If this header is passed by
+	// the client, its value will be overriden by the extension
+	// handler.
+	//
+	// Example:
+	//     Argocd-Target-Cluster-URL: "https://kubernetes.default.svc.cluster.local"
+	HeaderArgoCDTargetCluster = "Argocd-Target-Cluster-URL"
```

Comment:
Should this be added to the `### Headers` section of the docs?

Reply:
Done.